diff --git b/arch/arm/boot/dts/suniv-f1c100s-ns2009.dtsi b/arch/arm/boot/dts/suniv-f1c100s-ns2009.dtsi
new file mode 100644
index 0000000..c3b6615
--- /dev/null
+++ b/arch/arm/boot/dts/suniv-f1c100s-ns2009.dtsi
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR X11)
+/*
+ * Copyright 2022 Michal Wolowik <michwolo@gmail.com>
+ */
+
+#include "suniv-f1c100s.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+
+&i2c0 {
+	pinctrl-0 = <&i2c0_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+
+	ns2009: touchscreen@48 {
+		compatible = "nsiway,ns2009";
+		reg = <0x48>;
+		touchscreen-fuzz-x = <8>;
+		touchscreen-fuzz-y = <8>;
+	};
+};
+
+&pio {
+    ts_reset_pin: ts_reset_pin@0 {
+        pins = "PE9";
+        function = "gpio_out";
+    };
+};
