diff --git a/arch/arm/boot/dts/suniv-f1c100s.dtsi b/arch/arm/boot/dts/suniv-f1c100s.dtsi
index c059613..2e9596a 100644
--- a/arch/arm/boot/dts/suniv-f1c100s.dtsi
+++ b/arch/arm/boot/dts/suniv-f1c100s.dtsi
@@ -280,7 +280,7 @@
 		};
 
         usb_otg: usb@1c13000 {
-            compatible = "allwinner,suniv-musb";
+            compatible = "allwinner,suniv-f1c100s-musb";
             reg = <0x01c13000 0x0400>;
             clocks = <&ccu CLK_BUS_OTG>;
             resets = <&ccu RST_BUS_OTG>;
@@ -294,7 +294,7 @@
         };
 
         usbphy: phy@1c13400 {
-            compatible = "allwinner,suniv-usb-phy";
+            compatible = "allwinner,suniv-f1c100s-usb-phy";
             reg = <0x01c13400 0x10>;
             reg-names = "phy_ctrl";
             clocks = <&ccu CLK_USB_PHY0>;
